services/src/main/java/org/keycloak/forms/login/freemarker/FreeMarkerLoginFormsProvider.java

```diff
@@ -544,7 +549,16 @@ public Response createLoginWebAuthn() {
     }
 
     @Override
-    public Response createRegistration() {
+    public Response createRegistration(FormContext context) {
+        MultivaluedMap<String, String> params = context.getUriInfo().getQueryParameters();
```

Comment:
Same comment like for "createPasswordReset()" method above. Reading from authenticationSession will allow to remove "context" and avoid changing the signature of methods.